ethpm/contract.py

```diff
@@ -17,10 +17,9 @@ class LinkableContract(Contract):
 
     deployment_link_refs = []  # type: List[Dict[str, Any]]
     runtime_link_refs = []  # type: List[Dict[str, Any]]
-    is_bytecode_linked = False
 
     def __init__(self, address: bytes = None, **kwargs: Dict[str, Any]) -> None:
-        if self.has_linkable_bytecode() and not self.is_bytecode_linked:
+        if self.needs_bytecode_linking():
```

Comment:
I've found that non-property based boolean checks like this are often error prone in python because `if self.needs_bytecode_linking` will always return truthy (since `function` objects are considered *truthy* in python.  Maybe this should be `is_bytecode_fully_linked`?